staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -551,14 +551,18 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		if err != nil {
 			return nil, nil, err
 		}
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return nil, nil, err
+		}
 
 		// If AllowUnconditionalUpdate() is true and the object specified by
 		// the user does not have a resource version, then we populate it with
 		// the latest version. Else, we check that the version specified by
 		// the user matches the version of latest storage object.
 		resourceVersion, err := e.Storage.Versioner().ObjectResourceVersion(obj)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, invalidResourceVersion(qualifiedResource, name, accessor.GetResourceVersion(), fmt.Sprintf("Failed to parse resource version of: %v", err))
```

Comment:
what message does this produce? I expected to just pass err.Error() to invalidResourceVersion

Reply:
```
curl --q -k --cert /var/kube/kubecfg.crt -H "Content-Type: application/json-patch+json" -XPATCH --key /var/kube/kubecfg.key  'https://172.16.29.130:443/api/v1/namespaces/default/configmaps/extra-cfg'  -d '[{"op": "replace", "path": "/metadata/resourceVersion", "value": "abc"}]'
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "configmaps \"extra-cfg\" is invalid: metadata.resourceVersion: Invalid value: \"abc\": Failed to parse resource version: strconv.ParseUint: parsing \"abc\": invalid syntax",
  "reason": "Invalid",
  "details": {
    "name": "extra-cfg",
    "kind": "configmaps",
    "causes": [
      {
        "reason": "FieldValueInvalid",
        "message": "Invalid value: \"abc\": Failed to parse resource version: strconv.ParseUint: parsing \"abc\": invalid syntax",
        "field": "metadata.resourceVersion"
      }
    ]
  },
  "code": 422
}
```